scripts/beesd.in

```diff
@@ -134,7 +134,7 @@ fi
     fi
     if (( "$OLD_SIZE" != "$NEW_SIZE" )); then
         INFO "Resize db: $OLD_SIZE -> $NEW_SIZE"
-        [ -f "$BEESHOME/beescrawl.$UUID.dat" ] && rm "$BEESHOME/beescrawl.$UUID.dat"
+        [ -f "$BEESHOME/beescrawl.dat" ] && rm "$BEESHOME/beescrawl.dat"
```

Comment:
I don't think this is the intention of the script. The `if` is just there to remove the legacy file at resize time. We actually don't want to remove the current file on resize, the data in there may still be valid. Actually, I think the wrapper script came only after the removal of the legacy location.

Reply:
Maybe it should become an `mv` if the new file does not already exist.

Reply:
> The `if` is just there to remove the legacy file at resize time. We actually don't want to remove the current file on resize, the data in there may still be valid.

According to my understanding of how bees work, I'm not sure this would be very useful.  `beescrawl.dat` remembers how much of the filesystems bees has crawled. When truncating (shrinking) the hash table, we destroy some part of the data in it, causing bees to miss duplicate data covered by the destroyed hashes. When enlarging the hash table, hash lookups may also now miss (because the cell will no longer be in the expected bucket), and the extant buckets will now be unnecessarily full, containing cells which can never be reached.

> Actually, I think the wrapper script came only after the removal of the legacy location.

No, I made sure of the order when I wrote the commit message (which implies the order). 

- The bash script was added in a384cd976a8cedad5035e2636c7fd05f6c8c9834, on 2016-11-27.
- The UUID was removed in 06e111c229331e152656e1840646a862a4410503, on 2016-12-02.

> Maybe it should become an `mv` if the new file does not already exist.

That seems to be entirely orthogonal to the code here, which runs specifically when the configured size does not match the current file size.

Reply:
Legacy beescrawl-$uuid.dat became legacy in 2016.  Anyone who's still running bees from that era must be extraordinarily patient--bees was ~30x slower then due to some silly bugs.  I'm pretty sure that code will not be executed any more.

Reply:
bees doesn't yet support resizing the hash table--it only sort-of works, accidentally, for integer-multiple size changes because of properties of modulo arithmetic.  For all other resize cases, you need a specialized support tool or explicit bees support for resize (outlined in https://github.com/Zygo/bees/issues/44#issuecomment-357327920).  Without that, the amount of data preserved when a hash table is resized by `truncate` is negligible, so scanning has to start over from the beginning.

Overall, the PR looks good.  I'd use 'rm -f' instead of 'rm' (we don't want this script to pop up a prompt for any reason).

Reply:
>  I'd use 'rm -f' instead of 'rm' (we don't want this script to pop up a prompt for any reason).

As far as I can tell the only way to enable confirmations for `/bin/rm` is through a command-line option, which is usually done using an alias. User aliases won't be active in the script, so (other than some rogue `rm` being in front of the `PATH`) I don't see a circumstance in which this command would result in an interactive prompt.

Reply:
Yeah yeah, I'm slow... This was my draft I didn't send yet:

>> According to my understanding of how bees work, I'm not sure this would be very useful.
>
> @Zygo Do you have some deeper insight? I remember you writing that even after truncating, the file content can still be useful. But I was seeing some strange or buggy behavior with this, thus I'd tend to just approve this change: It was always better for me to purge the hash and crawler status file after resizing.

So, LGTM. Also, I only realized later that this is about the crawl state file, not the hash file. I think we should also purge the hash file during resize?

Reply:
```
$ touch /tmp/testf
$ chmod 444 /tmp/testf
$ rm /tmp/testf
rm: remove write-protected regular empty file '/tmp/testf'? n
$ type rm
rm is hashed (/bin/rm)
```

Reply:
```
$ type rm
rm is hashed (/bin/rm)
$ touch /tmp/testf
$ chmod 444 /tmp/testf
$ rm /tmp/testf
rm: remove write-protected regular empty file '/tmp/testf'?
```

There are a bunch of rules about when rm does and doesn't prompt.  I don't bother to remember these rules--I just use -f when I intend files to cease to exist without question.

`rm -f` also doesn't emit an error message if the file doesn't exist, so you can reduce the line down to just `rm -f beescrawl.dat` (without the [ -e ] check).

Reply:
@kakra Purging the hash file (i.e. unlinking or truncating to 0 followed by truncate with the target size) is the cleanest thing to do. 

If the hash table is at the low end of the useful size range it will not matter very much as the table is constantly flushed out with new data; however, if the hash table is at the high end of the useful size range, then all the useless but filled hash cells will make bees miss a lot of dupes for a long time.  So in one case it doesn't matter while at the other extreme it matters a lot.